Symfony/CS/Fixer/PSR2/SwitchCaseSpaceFixer.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Symfony\CS\Fixer\PSR2;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+class SwitchCaseSpaceFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        $caseTokens = $tokens->findGivenKind(T_CASE);
+
+        foreach ($caseTokens as $caseIndex => $caseToken) {
```

Comment:
Second reason is performance, as `findGivenKind` is one loop on whole collection, and then you do a second look on partial collection.

It would be great if `findGivenKind` would be a generator, but it is not.